src/search/landmarks/landmark_heuristic.cc

```diff
@@ -18,18 +18,58 @@ LandmarkHeuristic::LandmarkHeuristic(
     const shared_ptr<AbstractTask> &transform, bool cache_estimates,
     const string &description, utils::Verbosity verbosity)
     : Heuristic(transform, cache_estimates, description, verbosity),
+      initial_landmark_graph_has_cycle_of_natural_orderings(false),
       use_preferred_operators(use_preferred_operators),
       successor_generator(nullptr) {
 }
 
+/* TODO: We would prefer the following two functions to be implemented
+    somewhere else as more generic graph algorithms. */
+static bool depth_first_search_for_cycle_of_natural_orderings(
```

Comment:
I personally would put such static methods even before the constructor, but I don't know if we have coding conventions regarding this.

Reply:
Good question. I usually try to put things close to where they are used, but putting static functions on top of the file also makes sense to me. I don't have strong feelings about this, but would probably also need to go through all the files to make it consistent.